lua/mini/notify.lua

```diff
@@ -309,10 +308,8 @@ MiniNotify.make_notify = function(opts)
 
     local id = MiniNotify.add(msg, level_name, level_data.hl_group)
     vim.defer_fn(function() MiniNotify.remove(id) end, level_data.duration)
-  end
-  return function(msg, level)
-    if not vim.in_fast_event() then return notify(msg, level) end
-    vim.schedule(function() notify(msg, level) end)
+
+    return id
```

Comment:
If compliance with `vim.notify` is important I can remove it - it's not essential and the user can write custom replacement of `make_notify`.